### ISSUE TITLE
Documentation request regarding Spigot compatibility

### ISSUE THREAD

**Issue by brandondrew** [enebo/Purugin#54]
It would be very useful to know up-front whether Spigot's fork of craftbukkit is compatible with Purugin (and Purogo).  If this has been tested and the answer is known, it would be great if the person with that knowledge can update the README.  If it has not been tested then those of us who want to try it will know we're not wasting our time attempting to get it working (and can update the README afterward).

Also, it would be useful to know what extra steps, if any, are needed to get the Spigot fork working with this.

**Comment by ryantheleach** [comment]
I havn't tested it but nearly all plugins that respect the Bukkit API and don't access craftbukkit / minecraft internals / implementation specific knowledge are compatible with Spigots fork.

**Comment by enebo** [comment]
I was able to run all of the examples I tried a month or two ago with the exception of the cross module dependencies (locs_plus.rb and port.rb depend on each other).  I patched an alternate mechanism which works and just realized now I never pushed that change.  So I think Spigot for the most part works but I would not be surprised if there are more surprises.  I just need to get JRuby 9k out before I feel happy enough to work on this again.

Bottom line though is Spigot seems to work surprisingly well.

**Comment by enebo** [comment]
note from previous comment.  Pushed module dep fix in 2fc7ec6.

**Comment by brandondrew** [comment]
Awesome.  Thanks!

On Tue, Jul 7, 2015 at 5:43 PM, Thomas E Enebo <notifications@github.com>
wrote:

> note from previous comment. Pushed module dep fix in 2fc7ec6
> <https://github.com/enebo/Purugin/commit/2fc7ec6eb40c28041ef51ba73d6fa6dcb43347cb>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/enebo/Purugin/issues/54#issuecomment-119351449>.
>



-- 
Brandon Zylstra
brandon.zylstra@gmail.com

**Comment by enebo** [comment]
Ok I spent a little time to figure out where they have spigot maven stuff and I was able to build Purugin.  I was even able to plunk it into a an old spigot/minecract plugins dir and it started up.  So if you have mvn and a jdk you can type 'mvn' and see if the resulting purugin jar in target works on your install.

A baseline as to whether people other than me is seeing success and we can update the wiki and the README that we are back in business...

Last commit of fa91657 should have the updated build magic!

**Comment by enebo** [comment]
So I believe it is compatible and anything not working should be reported as a new issue.  If there is something wrong with any documentation can I ask someone to PR the changes?  I probably say it runs with Bukkit and that is still basically true.  I did not need to switch to specific SpgotMC APIs to get this to run.  I do not think I mention Spigot specifically though.
